components/telemetry-operator/controller/logpipeline/controller.go

```diff
@@ -204,3 +215,35 @@ func (r *Reconciler) updateLogPipelineStatus(ctx context.Context, name types.Nam
 	}
 	return nil
 }
+
+func (r *Reconciler) updateMetrics(allPipelines *telemetryv1alpha1.LogPipelineList) {
+	r.allLogPipelines.Set(float64(len(allPipelines.Items)))
+	r.unsupportedLogPipelines.Set(float64(countUnsupportedPluginUsages(allPipelines)))
+}
+
+// syncUnsupportedPluginsTotal checks if any LogPipeline defines a unsupported Filter or Output.
+func countUnsupportedPluginUsages(pipelines *telemetryv1alpha1.LogPipelineList) int {
+	count := 0
+	for i := range pipelines.Items {
+		if !pipelines.Items[i].DeletionTimestamp.IsZero() {
+			continue
+		}
+		if usesUnsupportedPlugin(&pipelines.Items[i]) {
+			count++
+		}
+	}
+
+	return count
+}
+
+func usesUnsupportedPlugin(pipeline *telemetryv1alpha1.LogPipeline) bool {
```

Comment:
Think you can use the method of LogPipeline `ContainsCustomPlugin`

Reply:
Agree, will do it